### PR TITLE
Utilise Eddsa & Public Keys - Part 1

### PR DESCRIPTION
**Pull request by biscuitdey**
# Description

Use Eddsa signatures in createBpiSubject and updateBpiSubject

## Related Issue

#759 

## Motivation and Context

Use multiple types of signatures 

## How Has This Been Tested

Running all existing tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

**Review by Therecanbeonlyone1969 — DISMISSED** [review]
LGTM

File: examples/bri-3/prisma/seed.ts [hunk 1]
```diff
@@ -23,17 +24,33 @@ async function main() {
...
-          },
+      name: 'BpiAdmin',
+      description: 'Internal Bpi Subject of this Bpi',
+      publicKeys: {
+        createMany: {
+          data: [
+            {
+              type: PublicKeyType.ECDSA,
+              value:
+                '0x044e851fa6118d0d33f11ebf8d4cae2a25dca959f06c1ab87b8fec9ccbf0ca0021b7efc27c786f9480f9f11cfe8df1ae991329654308611148a35a2277ba5909fe',
```
Comment (ognjenkurtic) [inline examples/bri-3/prisma/seed.ts:35]: these two are the same? This is probably not intended as they are of different type?

File: examples/bri-3/src/bri/identity/bpiSubjects/capabilities/getAllBpiSubjects/getAllBpiSubjectsQuery.handler.ts [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
-import { Mapper } from '@automapper/core';
+import { PrismaMapper as Mapper } from '../../../../../shared/prisma/prisma.mapper';
 
 @QueryHandler(GetAllBpiSubjectsQuery)
 export class GetAllBpiSubjectsQueryHandler
   implements IQueryHandler<GetAllBpiSubjectsQuery>
 {
   constructor(
-    @InjectMapper() private autoMapper: Mapper,
+    private readonly mapper: Mapper,
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/identity/bpiSubjects/capabilities/getAllBpiSubjects/getAllBpiSubjectsQuery.handler.ts:12]: Please revert this, Prisma mapper should be used only in the storage level, so between prisma models and domain object

File: examples/bri-3/src/bri/identity/bpiSubjects/capabilities/getBpiSubjectById/getBpiSubjectByIdQuery.handler.ts [hunk 3]
```diff
@@ -4,16 +4,15 @@ import { BpiSubjectStorageAgent } from '../../agents/bpiSubjectsStorage.agent';
...
+import { PrismaMapper as Mapper } from '../../../../../shared/prisma/prisma.mapper';
 import { BpiSubject } from '../../models/bpiSubject';
 
 @QueryHandler(GetBpiSubjectByIdQuery)
 export class GetBpiSubjectByIdQueryHandler
   implements IQueryHandler<GetBpiSubjectByIdQuery>
 {
   constructor(
-    @InjectMapper() private autoMapper: Mapper,
+    private readonly mapper: Mapper,
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/identity/bpiSubjects/capabilities/getBpiSubjectById/getBpiSubjectByIdQuery.handler.ts:15]: Same here

File: examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts [hunk 4]
```diff
@@ -1,18 +1,17 @@
...
 import { UpdateBpiSubjectCommand } from './updateBpiSubject.command';
+import { PublicKey, PublicKeyType } from '../../models/publicKey';
 
 @CommandHandler(UpdateBpiSubjectCommand)
 export class UpdateBpiSubjectCommandHandler
   implements ICommandHandler<UpdateBpiSubjectCommand>
 {
   constructor(
-    @InjectMapper() private mapper: Mapper,
+    private readonly mapper: Mapper,
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts:14]: Same here

File: examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts [hunk 5]
```diff
@@ -23,17 +22,39 @@ export class UpdateBpiSubjectCommandHandler
...
+          publicKeyType,
+          key.value,
+          bpiSubjectToUpdate.id,
+        );
+
+        return newKey;
+      }),
+    );
+
     this.agent.updateBpiSubject(
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts:48]: Above code should be hidden behind this method. remmember, code in command handlers explains WHAT happens, not HOW.

**Comment by ognjenkurtic** [comment]
Please also run and fix prettier and linter

File: examples/bri-3/prisma/seed.ts (hunk 1 above)
Comment (biscuitdey) [reply examples/bri-3/prisma/seed.ts:35]: Done.

File: examples/bri-3/src/bri/identity/bpiSubjects/capabilities/getAllBpiSubjects/getAllBpiSubjectsQuery.handler.ts (hunk 2 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/identity/bpiSubjects/capabilities/getAllBpiSubjects/getAllBpiSubjectsQuery.handler.ts:12]: Done.

File: examples/bri-3/src/bri/identity/bpiSubjects/capabilities/getBpiSubjectById/getBpiSubjectByIdQuery.handler.ts (hunk 3 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/identity/bpiSubjects/capabilities/getBpiSubjectById/getBpiSubjectByIdQuery.handler.ts:15]: Done.

File: examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts (hunk 4 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts:14]: Done.

File: examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts (hunk 5 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts:48]: Fixed.

**Comment by biscuitdey** [comment]
Code added to #794

**Comment by biscuitdey** [comment]
Code added to #794

**Pull request closed without merging**